packages/editor/src/components/collab-sidebar/index.js

```diff
@@ -0,0 +1,303 @@
+	const { getEntityRecords } = resolveSelect( coreStore );
+
+	// eslint-disable-next-line @wordpress/data-no-store-string-literals
+	const { openGeneralSidebar } = useDispatch( 'core/edit-post' );
+	const [ blockCommentID, setBlockCommentID ] = useState( null );
+	const [ showCommentBoard, setShowCommentBoard ] = useState( false );
+
+	const [ threads, setThreads ] = useState( () => [] );
+	const { postId } = useSelect( ( select ) => {
+		return {
+			postId: select( editorStore ).getCurrentPostId(),
+		};
+	}, [] );
+
+	const clientId = useSelect( ( select ) => {
+		const { getSelectedBlockClientId } = select( blockEditorStore );
+		setBlockCommentID(
+			select( blockEditorStore ).getBlock( getSelectedBlockClientId() )
+				?.attributes.blockCommentId
+		);
+		return getSelectedBlockClientId();
+	}, [] );
+
+	// Get the dispatch functions to save the comment and update the block attributes.
+	const { updateBlockAttributes } = useDispatch( blockEditorStore );
+
+	const openCollabBoard = () => {
+		setShowCommentBoard( true );
+		openGeneralSidebar( 'edit-post/collab-sidebar' );
+	};
+
+	// Function to save the comment.
+	const addNewComment = async ( comment, parentCommentId ) => {
+		const args = {
+			post: postId,
+			content: comment,
+			comment_type: 'block_comment',
+			comment_approved: 0,
+		};
+
+		if ( parentCommentId ) {
+			args.parent = parentCommentId;
+		}
+
+		const savedRecord = await saveEntityRecord( 'root', 'comment', args );
+
+		if ( savedRecord ) {
+			// If it's a main comment, update the block attributes with the comment id.
+			if ( ! parentCommentId ) {
+				updateBlockAttributes( clientId, {
+					blockCommentId: savedRecord?.id,
+				} );
+			}
+
+			createNotice(
+				'snackbar',
+				parentCommentId
+					? // translators: Reply added successfully
+					  __( 'Reply added successfully.' )
+					: // translators: Comment added successfully
+					  __( 'Comment added successfully.' ),
+				{
+					type: 'snackbar',
+					isDismissible: true,
+				}
+			);
+			fetchComments();
+		} else {
+			onError();
+		}
+	};
+
+	const onCommentResolve = async ( commentId ) => {
+		const savedRecord = await saveEntityRecord( 'root', 'comment', {
+			id: commentId,
+			status: 'approved',
+		} );
+
+		if ( savedRecord ) {
+			// translators: Comment resolved successfully
+			createNotice( 'snackbar', __( 'Thread marked as resolved.' ), {
+				type: 'snackbar',
+				isDismissible: true,
+			} );
+
+			fetchComments();
+		} else {
+			onError();
+		}
+	};
+
+	const onEditComment = async ( commentId, comment ) => {
+		const savedRecord = await saveEntityRecord( 'root', 'comment', {
+			id: commentId,
+			content: comment,
+		} );
+
+		if ( savedRecord ) {
+			createNotice(
+				'snackbar',
+				// translators: Comment edited successfully
+				__( 'Thread edited successfully.' ),
+				{
+					type: 'snackbar',
+					isDismissible: true,
+				}
+			);
+
+			fetchComments();
```

Comment:
These async functions will end their execution potentially before `fetchComments()` has finished. Is that intended? Do we mean to wait for comment fetching?